persistence/src/commonTest/kotlin/com/wire/kalium/persistence/dao/UserDAOTest.kt

```diff
@@ -251,11 +251,26 @@ class UserDAOTest : BaseDatabaseTest() {
             )
             db.userDAO.insertUsers(mockUsers)
             //when
-            val searchResult = db.userDAO.getUserByNameOrHandleOrEmail(commonPrefix).first()
+            val searchResult = db.userDAO.getUserByNameOrHandleOrEmailAndConnectionState(commonPrefix,UserEntity.ConnectionState.ACCEPTED).first()
             //then
             assertEquals(mockUsers, searchResult)
         }
 
+    @Test
+    fun givenAExistingUsers_whenQueried_ThenResultsUsersAreConnected() = runTest {
+        //given
+        val commonNamePrefix = "commonName"
+
+        val mockUsers = listOf(USER_ENTITY_1, USER_ENTITY_2, USER_ENTITY_3)
+        db.userDAO.insertUsers(mockUsers)
+        //when
+        val searchResult = db.userDAO.getUserByNameOrHandleOrEmailAndConnectionState(commonNamePrefix, UserEntity.ConnectionState.ACCEPTED).first()
+        //then
+        searchResult.forEach { userEntity ->
+            assertEquals(UserEntity.ConnectionState.ACCEPTED,userEntity.connectionStatus)
+        }
+    }
```

Comment:
Could you insert a couple of users that are **not** connected and just one conected, making sure that is the one that will be returned?

Reply:
sure ,good idea